### DISCUSSION TITLE
Check if there is an available audio input source

### DISCUSSION THREAD

**Discussion by hongjiapeng** [naudio/NAudio#939] [Q&A]
Can Naudio detect if the system has an available audio input source?

**Comment by Mauricio-MN** [comment]
Here are some examples:

```
var capCount = WaveOut.DeviceCount;
var recCount = WaveIn.DeviceCount;

var enumerator = new MMDeviceEnumerator();

var[] enumsRecordDevice = enumerator.EnumerateAudioEndPoints(DataFlow.Capture, DeviceState.Active);
var[] enumsCaptureDevice = enumerator.EnumerateAudioEndPoints(DataFlow.Render, DeviceState.Active);
```
I recommend taking a closer look at the author's documentation and/or blog.
This answer was based on another one from stackoverflow: [](https://stackoverflow.com/questions/1449136/enumerate-recording-devices-in-naudio)
